tensorflow/core/kernels/lookup_table_op.cc

```diff
@@ -723,9 +852,10 @@ class MutableDenseHashTable final : public LookupInterface {
     num_entries_ = 0;
 
     const int64 key_size = key_shape_.num_elements();
+    const int64 allocate_size = key_bytes_ / sizeof(K);
     Tensor* key_buckets_tensor;
     TF_RETURN_IF_ERROR(ctx->allocate_persistent(
-        key_dtype(), TensorShape({num_buckets_, key_size}), &key_buckets_,
+        key_dtype(), TensorShape({num_buckets_, allocate_size}), &key_buckets_,
```

Comment:
Its unclear to me that when use_value_ptr_ is false also we are modifying key_buckets_?